### PR TITLE
Sprint: update src/marine/gds2_sst2ioda.py code for JEDI Data Conventions

### LINKED ISSUE #935: Upgrade converter for gds2_sst2ioda.py to follow the JEDI Data Conventions
Update the Python converter ```src/marine/gds2_sst2ioda.py``` to follow the JEDI Data Conventions.

- Update variable names
- Ensure variable units match the units in the conventions table
- Update the global attributes dictionary as needed
- Run the converter test to make sure it runs (but will fail because of unupdated test reference)
- Run ```ioda-validate.x``` on your output file and verify that there are no errors or warnings
- Update test reference file
- Run unit tests making use of the target converter and verify that they pass
- Make sure you verify the results by examining the file using ```ncdump -t```, ```h5dump```, ```HDFview```, or a similar application.

### PR DESCRIPTION
**Pull request by gthompsnJCSDA**
## Description

This PR updates the ```src/marine/gds2_sst2ioda.py``` script to follow the JEDI Data Conventions.  Besides some of the regular changes like ```datetime``` to ```dateTime``` a number of relatively minor improvements to code structure becoming more generic and similar to many other marine converters.

### Issue(s) addressed

Closes #935 

## Acceptance Criteria (Definition of Done)

Success on the 2 ctests of this script (2 different processing levels of input data) and ioda-validate program along with positive reviews.

## Dependencies

Waiting on the following PRs:
- [ ] waiting on [JCSDA/ioda/pull/704](https://github.com/JCSDA-internal/ioda/pull/704)

## Impact

No impacts if kept within ```feature/sprint-ioda-converters```

### CODE REVIEW

File: src/marine/gds2_sst2ioda.py [hunk 1]
```diff
@@ -16,6 +16,7 @@
 import numpy as np
 from multiprocessing import Pool
 import os
+import re
```
Comment (fabiolrdiniz) [inline src/marine/gds2_sst2ioda.py:19]:
```suggestion
```

**Review by fabiolrdiniz — APPROVED** [review]
Hi, @gthompsnJCSDA. Thanks for adding me here. I assume the conversion (Kelvin to Celsius) is not needed anymore, right?

**Review by BenjaminRuston — APPROVED** [review]
works

**Review by HamidehGMAO — APPROVED** [review]
Looks fine to me. Thanks. I'm just wondering why other unrelated tests are failing on AWS?!

**Comment by srherbener** [comment]
The relevant tests are passing. I'll merge this in now. Thanks!

**Comment by srherbener** [comment]
> Looks fine to me. Thanks. I'm just wondering why other unrelated tests are failing on AWS?!

Good question. My understanding is that we would have test failures on the feature/sprint-ioda-converters branch until all scripts had been properly upgraded to the new JEDI Conventions. I believe that the remaining CI test failures are just saying we aren't finished yet with all of the upgrades.

**Pull request merged**